server/node_filter.go

```diff
@@ -436,7 +435,7 @@ func (nf *NodeFilter) KubeletOutdatedNodes() (nodes []*cke.Node) {
 			log.Warn("kubelet's container runtime cannot be changed", nil)
 		case cke.KubernetesImage.Name() != st.Image:
 			fallthrough
-		case !reflect.DeepEqual(currentConfig, runningConfig):
+		case !cmp.Equal(currentConfig, runningConfig):
```

Comment:
I use `cmp.Equal instead` of `reflect.DeepEqual`.

Because `reflect.DeepEqual` has got false positives after upgrading k8s 1.23.
And the mtest never succeed with following log.

```
2022-08-02T09:25:53.7562730Z [31mAug 02 09:18:45 host1 docker[1326]: 2022-08-02T09:18:45.409510Z host1 cke debug: "kubelet outdated" config="&TypeMeta***Kind:,APIVersion:,***" current_builtin_args="[kubelet --config=/etc/kubernetes/kubelet/config.yml --kubeconfig=/etc/kubernetes/kubelet/kubeconfig --hostname-override=10.0.0.101 --network-plugin=cni --container-runtime=remote --container-runtime-endpoint=unix:///var/run/k8s-containerd.sock]" current_builtin_env="map[]" current_extra_args="[]" current_extra_env="map[]" diff="" node="10.0.0.101" st_builtin_args="[kubelet --config=/etc/kubernetes/kubelet/config.yml --kubeconfig=/etc/kubernetes/kubelet/kubeconfig --hostname-override=10.0.0.101 --network-plugin=cni --container-runtime=remote --container-runtime-endpoint=unix:///var/run/k8s-containerd.sock]" st_builtin_env="map[]" st_extra_args="[]" st_extra_env="map[]"
```